### ISSUE TITLE
[FAIL] Implement Local Storage

### ISSUE THREAD

**Issue by Gamagu** [SE-TINF22B2/G1-CashCompass#171]
The local storage implements the DataAdapter. The Data adapter is the interface which defines what a data provider needs to expose, that the serialiser can access the data in a unified way.

Until now, 3 Methods are required: 
- The load method is for retrieving objects from the database. Those objects are identified by a path and an id. Due to the usage of generics, the method calls a factory and we need to implement this only once.
- The store method is also generic and needs only one implementation because every object has a serialiser.
- And a initialLoad method which reads all data on app startup.

Things to pay attention:
- how the id is stored: the field "id" returns the localId in case the object is not present on the server, e.g. right after the initialisation and in offline mode, when the object exists on the server, the remote id is retuned
- The local id needs to be initialized manually. 
- there must be a function that can delete objects, because the copy identified by the local id should be deleted after the sync.

**Comment by stefanrudi2004** [comment]
We had a meeting to discuss the topic last week. 
Time: 1 h

**Comment by Gamagu** [comment]
i renamed the corresponding file for this class. This change is already merged into main, if youre already working on this feature, a rebase should fit this
